src/ui/components/ExperimentBrowserPage.tsx

```diff
@@ -24,13 +24,146 @@ enum Environment {
   STAGE = "stage",
 }
 
+type ToastVariant = "success" | "danger";
+
+interface AddToastParams {
+  message: string;
+  variant: ToastVariant;
+  autohide?: boolean;
+}
+
+async function tryEnroll(
+  experiment: NimbusExperiment,
+  branchSlug: string,
+): Promise<AddToastParams> {
+  try {
+    const enrolled = await browser.experiments.nimbus.forceEnroll(
+      experiment,
+      branchSlug,
+    );
+    if (enrolled) {
+      return { message: "Enrollment successful", variant: "success" };
+    } else {
+      return { message: "Enrollment failed", variant: "danger" };
+    }
+  } catch (error) {
+    return {
+      message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
+      variant: "danger",
+    };
+  }
+}
+
+async function tryGenerateTestId(
+  experiment: NimbusExperiment,
+  branchSlug: string,
+): Promise<AddToastParams> {
+  try {
+    const result = await browser.experiments.nimbus.generateTestIds(
+      experiment,
+      branchSlug,
+    );
+    if (result) {
+      await navigator.clipboard.writeText(result);
+      return {
+        message: `Id copied to clipboard. Test Id: ${result}`,
+        variant: "success",
+        autohide: false,
+      };
+    } else {
+      return { message: "Test Id generation failed", variant: "danger" };
+    }
+  } catch (error) {
+    return {
+      message: `Error generating test Id: ${(error as Error).message ?? String(error)}`,
+      variant: "danger",
+    };
+  }
+}
+
+const ExperimentRow: FC<{ experiment: NimbusExperiment }> = ({
```

Comment:
Can we put this below ExperimentBrowserPage? it makes more sense when youre reading the file to have the main thing first and the details later.